src/test/java/org/betonquest/betonquest/api/logger/util/LogValidator.java

```diff
@@ -0,0 +1,109 @@
+    }
+
+    @Override
+    public void flush() {
+        records.clear();
+    }
+
+    @Override
+    public void close() {
+        // Empty
+    }
+
+    /**
+     * Assert that the {@link LogValidator} does not have any LogRecord left to check.
+     */
+    public void assertEmpty() {
+        Assertions.assertTrue(records.isEmpty(), "The records list should be empty!");
+    }
+
+    /**
+     * Assert that the level and the message and the throwable class of the LogRecord are equal.
+     *
+     * @param level            The expected level.
+     * @param message          The expected message.
+     * @param throwable        The expected throwable class.
+     * @param throwableMessage The expected throwable message.
+     */
+    public void assertLogEntry(final Level level, final String message, final Class<? extends Throwable> throwable, final String throwableMessage) {
+        final LogRecord record = records.pop();
+        assertNotNull(record, "The record is unexpected null!");
+        assertEntry(record, level, message);
+        assertEntry(record, throwable);
+        assertEntry(record, throwableMessage);
+    }
+
+    /**
+     * Assert that the level and the message and the throwable class of the LogRecord are equal.
+     *
+     * @param level     The expected level.
+     * @param message   The expected message.
+     * @param throwable The expected throwable class.
+     */
+    public void assertLogEntry(final Level level, final String message, final Class<? extends Throwable> throwable) {
+        final LogRecord record = records.pop();
+        assertNotNull(record, "The record is unexpected null!");
+        assertEntry(record, level, message);
+        assertEntry(record, throwable);
+    }
+
+    /**
+     * Assert that the level and the message of the LogRecord are equal.
+     *
+     * @param level   The expected level.
+     * @param message The expected message.
+     */
+    public void assertLogEntry(final Level level, final String message) {
+        final LogRecord record = records.pop();
+        assertNotNull(record, "The record is unexpected null!");
+        assertEntry(record, level, message);
+    }
+
+    private void assertEntry(final LogRecord record, final String throwableMessage) {
+        assertEquals(throwableMessage, record.getThrown().getMessage(), "Expected log throwable message does not equal!");
+    }
+
+    private void assertEntry(final LogRecord record, final Class<? extends Throwable> throwable) {
+        assertNotNull(record.getThrown(), "Expected log throwable is null!");
+        assertEquals(throwable, record.getThrown().getClass(), "Expected log throwable does not equal!");
+    }
+
+    private void assertEntry(final LogRecord record, final Level level, final String message) {
+        assertEquals(level, record.getLevel(), "Expected log level does not equal!");
+        assertEquals(message, record.getMessage(), "Expected log message does not equal!");
+    }
```

Comment:
These methods contain duplicated code. They can be merged by calling each other:
```suggestion
    /**
     * Assert that the level and the message and the throwable class of the LogRecord are equal.
     *
     * @param level            The expected level.
     * @param message          The expected message.
     * @param throwable        The expected throwable class.
     * @param throwableMessage The expected throwable message.
     */
    public void assertLogEntry(final Level level, final String message, final Class<? extends Throwable> throwable, final String throwableMessage) {
        final LogRecord record = records.pop();
        assertEntry(record, level, message, throwable, throwableMessage);
    }

    /**
     * Assert that the level and the message and the throwable class of the LogRecord are equal.
     *
     * @param level     The expected level.
     * @param message   The expected message.
     * @param throwable The expected throwable class.
     */
    public void assertLogEntry(final Level level, final String message, final Class<? extends Throwable> throwable) {
        final LogRecord record = records.pop();
        assertEntry(record, level, message, throwable);
    }

    /**
     * Assert that the level and the message of the LogRecord are equal.
     *
     * @param level   The expected level.
     * @param message The expected message.
     */
    public void assertLogEntry(final Level level, final String message) {
        final LogRecord record = records.pop();
        assertEntry(record, level, message);
    }

    private void assertEntry(final LogRecord record, final Level level, final String message, final Class<? extends Throwable> throwable, final String throwableMessage) {
        assertEntry(record, level, message, throwable);
        assertEquals(throwableMessage, record.getThrown().getMessage(), "Expected log throwable message does not equal!");
    }

    private void assertEntry(final LogRecord record, final Level level, final String message, final Class<? extends Throwable> throwable) {
        assertEntry(record, level, message);
        assertNotNull(record.getThrown(), "Expected log throwable is null!");
        assertEquals(throwable, record.getThrown().getClass(), "Expected log throwable does not equal!");
    }

    private void assertEntry(final LogRecord record, final Level level, final String message) {
        assertNotNull(record, "The record is unexpected null!");
        assertEquals(level, record.getLevel(), "Expected log level does not equal!");
        assertEquals(message, record.getMessage(), "Expected log message does not equal!");
    }
```
The `records.pop()` call might be inlined if so desired.